pyHalo/realization_extensions.py

```diff
@@ -497,3 +502,201 @@ def _get_fluctuation_halos(realization, fluctuation_amplitude, fluctuation_size,
+            
+            NCC_num = (term_1 - ((term_2_1*term_2_2)/term_3))/Area
+            
+            term_4 = simps(simps((kappa_interp_1**2)*mask_interp_2, x_, axis=0), y_, axis=-1)
+            term_5 = term_2_1**2
+            term_6 = term_3
+            
+            NCC_den_1 = (term_4 - (term_5/term_6))/Area
+            
+            term_7 = simps(simps(mask_interp_1*(kappa_interp_2**2), x_, axis=0), y_, axis=-1)
+            term_8 = term_2_2**2
+            term_9 = term_3
+            
+            NCC_den_2 = (term_7 - (term_8/term_9))/Area   
+            
+            if normalization == True:
+                corr[i,j] = NCC_num/np.sqrt(NCC_den_1*NCC_den_2) 
+            else:
+                corr[i,j] = NCC_num
+            
+    end_time = time.time()
+    
+    print(f"It took {end_time-start_time:.2f} seconds to compute the correlation map") 
+    
+    return corr
+
+def xi_l(l, corr, r, mu):
+    """
+    This function computes the multipoles of the two-point correlation function.
+    
+    :param l: the order of the multipole, E.g., l=0 :monopole, l=1: dipole, l=2: quadrupole, etc.
+    :param corr: the two-point correlation function on the (mu, r) coordinate grid
+    :param r: an array of uniformly logarithmically spaced separations of interest
+    :param mu: an array of cosines of the rotation angles of vector r. E.g., mu = np.linspace(-1, 1, 100) contains the cosines 
+        of the angles between 0 and 180 degrees.
+    :return: r and the two-point correlation function multipole of order l
+    """
+    
+    T_l = eval_chebyt(l, mu)  
+    func = corr*T_l
+    
+    if l==0:
+        prefactor = 1/np.pi
+    else:
+        prefactor = 2/np.pi
+    
+    xi_l = np.zeros(r.shape[0])
+    for i in range(r.shape[0]):
+        xi_l[i] = simps(func[i], np.flip(np.arccos(mu)), axis=0)
+               
+    return r, prefactor*xi_l 
+    
+def xi_l_to_Pk_l(r, xi_l, l=0, extrapolate=True):
+
+    """
+    This function translates the correlation multipoles to power spectrum multipoles using Hankel Transform.
+    
+    :param r: an array of uniformly logarithmically spaced separations of interest
+    :param xi_l: the two-point correlation function multipole of order l
+    :param l: the order of the multipole, E.g., l=0 :monopole, l=1: dipole, l=2: quadrupole, etc.
+    :param corr: the two-point correlation function on the (mu, r) coordinate grid
+    :param extrapolate: if true extrapolates the power spectrum multipole with a power law to improve
+        the smoothness
+    :return: wavenumber (k) and the power spectrum multipole of order l
+    
+    """  
+    list_zero = list(np.where(r==0)[0])
+    r = np.delete(r, list_zero)
+    xi_l = np.delete(xi_l, list_zero)
+
+    prefactor = 2*np.pi*(-1j)**l
+    H = Hankel(r, nu=l, lowring = True)
+    k, Pk_l_ = H(xi_l, extrap=extrapolate)
+    
+    Pk_l = Pk_l_*prefactor
+     
+    return k, Pk_l.real
+
+def fitting_func(r, xi_l, r_min, r_max):
```

Comment:
please give a more descriptive name to this function, something like "correlation_function_statistics"